spring-integration-core/src/main/java/org/springframework/integration/config/DefaultConfiguringBeanFactoryPostProcessor.java

```diff
@@ -172,18 +174,21 @@ private void registerMessagePublishingErrorHandler() {
 	private void registerNullChannel() {
 		if (this.beanFactory.containsBean(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME)) {
 			BeanDefinition nullChannelDefinition = null;
-			if (this.beanFactory.containsBeanDefinition(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME)) {
-				nullChannelDefinition =
-						this.beanFactory.getBeanDefinition(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME);
-			}
-			else {
-				BeanDefinitionRegistry parentBeanFactory =
-						(BeanDefinitionRegistry) this.beanFactory.getParentBeanFactory();
-				if (parentBeanFactory != null) {
-					nullChannelDefinition =
-							parentBeanFactory.getBeanDefinition(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME);
+			BeanFactory beanFactory = this.beanFactory;
+			do {
+				if (beanFactory instanceof ConfigurableListableBeanFactory) {
+					ConfigurableListableBeanFactory listable = (ConfigurableListableBeanFactory) beanFactory;
+					if (listable.containsBeanDefinition(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME)) {
+						nullChannelDefinition = listable
+								.getBeanDefinition(IntegrationContextUtils.NULL_CHANNEL_BEAN_NAME);
+					}
 				}
-			}
+				// don't think it will hurt to always find the parent
```

Comment:
I'm not sure that we really are going to accept so internal thoughts as comments.
Why can't we really live without these comments at all?

Reply:
Agreed and no problem

Reply:
I guess I was just explaining my thinking